### PR TITLE
Add table-hover-color variables

### LINKED ISSUE #26465: v4 $table-hover-color option needed
hiya, busy theming bootstrap for my app. previously i did something like add my own class to a table and theme that class but this time im using the variables to get bootstrap itself to be themed. 

as is now:
````
$table-cell-padding:          .75rem !default;
$table-cell-padding-sm:       .3rem !default;

$table-bg:                    transparent !default;
$table-accent-bg:             rgba($black, .05) !default;
$table-hover-bg:              rgba($black, .075) !default;
$table-active-bg:             $table-hover-bg !default;

$table-border-width:          $border-width !default;
$table-border-color:          $gray-300 !default;

$table-head-bg:               $gray-200 !default;
$table-head-color:            $gray-700 !default;

$table-dark-bg:               $gray-900 !default;
$table-dark-accent-bg:        rgba($white, .05) !default;
$table-dark-hover-bg:         rgba($white, .075) !default;
$table-dark-border-color:     lighten($gray-900, 7.5%) !default;
$table-dark-color:            $body-bg !default;

$table-striped-order:         odd !default;

$table-caption-color:         $text-muted !default;
````

i propose the addition of 

````
$table-hover-color: $body-color !default;
````

as soon as you make the hover a darker colour for instance  being able to change the fonts colour is quite important. 

its easy enough to add into my custom css but it might break stuff regarding the different primary, secondary, info etc context classes

changes needed:
add `$table-hover-color: $body-color !default;` to _variables.scss
add `color:$table-hover-color` to _tables.scss line 93

bootstrap 4.1

### PR DESCRIPTION
**Pull request by mdo**
Fixes #26465.

Strictly speaking, this might constitute new features with the new variables, so we may need to do this in v4.2 or a v4.3. Lemme know your thoughts, @twbs/css-review.

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -352,8 +352,10 @@ $hr-margin-y:                 $spacer !default;
 $table-cell-padding:          .75rem !default;
 $table-cell-padding-sm:       .3rem !default;
 
+$table-color:                 $body-color !default;
```
Comment (MartijnCuppens) [inline scss/_variables.scss]: Where do you use this variable?

Comment (XhmikosR) [reply scss/_variables.scss]: And if it's unused why doesn't fusv catch it :P

Comment (mdo) [reply scss/_variables.scss]: This was moved up from farther down the file.

Comment (MartijnCuppens) [reply scss/_variables.scss]: This looks like a new variable, `$table-color` is not used in `v4-dev`. And indeed, very strange `fusv` doesn't catch this. No idea what's going on here.

Comment (XhmikosR) [reply scss/_variables.scss]: I think it's used in the sense it's used once, to set the `$table-hover-color` hence why fusv doesn't catch this.

Comment (mdo) [reply scss/_variables.scss]: Ah I see what you mean now, I assumed it was already in use within `_tables.scss`. We can add it there though.

**Review by MartijnCuppens — APPROVED** [review]

**Pull request merged**